packages/location-management/src/components/EditLocationUnit/index.tsx

```diff
@@ -0,0 +1,162 @@
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const [isJurisdiction, setIsJurisdiction] = useState<boolean>(true);
+  const { broken, errorMessage, handleBrokenPage } = useHandleBrokenPage();
+  const user = useSelector((state) => getUser(state));
+
+  // location being edited id
+  const { id: locId } = props.match.params;
+
+  const thisLocation = useSelector((state) => getLocationUnitById(state, locId)) ?? undefined;
+  const [loading, setLoading] = useState<boolean>(!thisLocation);
+
+  React.useEffect(() => {
+    // get location; we are making 2 calls to know if location is a jurisdiction or a structure
+    const commonParams = {
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      return_geometry: true,
+    };
+    const structureParams = {
+      ...commonParams,
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      is_jurisdiction: false,
+    };
+    const jurisdictionParams = {
+      ...commonParams,
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      is_jurisdiction: true,
+    };
+
+    const locationsDispatcher = (location: LocationUnit | null) => {
+      if (location) {
+        const locations = [location];
+        dispatch(fetchLocationUnits(locations));
+      }
+    };
+    // asynchronously get jurisdiction as structure and jurisdiction, depending on the resolved
+    // promise, we can then know if the location to edit is a jurisdiction or structure
+    const firstPromise = loadJurisdiction(
+      locId,
+      undefined,
+      openSRPBaseURL,
+      jurisdictionParams,
+      service
+    )
+      .then((res) => {
+        if (res) {
+          locationsDispatcher(res);
+        }
+      })
+      .catch((err) => {
+        throw err;
+      });
+    const secondPromise = loadJurisdiction(
+      locId,
+      undefined,
+      openSRPBaseURL,
+      structureParams,
+      service
+    )
+      .then((res) => {
+        if (res) {
+          setIsJurisdiction(false);
+          locationsDispatcher(res);
+        }
+      })
+      .catch((err) => {
+        throw err;
+      });
+    Promise.all([firstPromise, secondPromise])
+      .catch((err) => handleBrokenPage(err))
+      .finally(() => {
+        setLoading(false);
+      });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [locId]);
+
+  if (loading) {
+    return <Spin size="large"></Spin>;
+  }
+
+  if (broken) {
+    return <BrokenPage errorMessage={errorMessage} />;
+  }
+
+  const initialValues = getLocationFormFields(thisLocation, instance, isJurisdiction);
+  const cancelHandler = () => history.push(redirectAfterAction);
+
+  const locationFormProps = {
+    initialValues,
+    redirectAfterAction,
+    hidden,
+    disabled,
+    onCancel: cancelHandler,
+    service,
+    openSRPBaseURL,
+    user: user.username,
+  };
+  const pageTitle = `${EDIT_LOCATION_UNIT} | ${thisLocation?.properties.name}`;
```

Comment:
I might have seen a discussion on this before that the page title should be a constant string e.g Edit Location and not the location name. The location name should instead go to the page header? cc @dubdabasoduba

Reply:
what's the argument against having the name in the url?, IMO, the bonus here is that the document title reflects the pages title as accurately as possible. cc @dubdabasoduba

Reply:
I assume we are talking about adding this to the <title> element on HTML right? We never had that requirement. If we do this we might want to do it for all edit pages in the platform.

Reply:
cool.